packages/core/src/modules/ledger/services/LedgerService.ts

```diff
@@ -110,6 +111,34 @@ export class LedgerService {
     return this._poolHandle
   }
 
+  public async registerPublicDid(submitterDid: Did, targetDid: Did, verkey: string, alias: string, role?: NymRole) {
+    try {
+      this.logger.debug(`Register public did on ledger '${targetDid}'`)
+
+      const request = await this.indy.buildNymRequest(submitterDid, targetDid, verkey, alias, role || null)
+
+      const response = await this.submitWriteRequest(request, submitterDid)
+
+      this.logger.debug(`Registered public did '${targetDid}' on ledger`, {
+        response,
+      })
+
+      return targetDid
```

Comment:
One final nitpick: Maybe we shouldn't return the targetDid? It's already an input parameter, so it doesn't provide extra context to the method caller to return it. WDYT?

Reply:
Yes, you are right about that. Actually I left it for consistency with other methods from ledger service such as `registerSchema` and `registerCredentialDefinition`. Do you think it'd better to return void?

In that way, in the registerPublicDid test I can do a getPublicDid right after the DID registration (something I don't want to do because that that method is tested in a separate test, but probably should be the unique 'good' way to test it).

Reply:
I think it is fine for now. We can tackle this in a separate PR and overhaul all methods